sequencer/txtracker.go

```diff
@@ -13,46 +13,84 @@ import (
+	Cost                   *big.Int       // Cost = Amount + Benefit
+	Benefit                *big.Int       // GasLimit * GasPrice
+	IsClaim                bool           // Needed to calculate efficiency
+	BatchResources         batchResources // To check if it fits into a batch
+	Efficiency             float64
+	RawTx                  []byte
+	constraints            batchConstraintsFloat64
+	weightMultipliers      batchResourceWeightMultipliers
+	resourceCostMultiplier float64
+	totalWeight            float64
+	ReceivedAt             time.Time // To
 }
 
-// newTxTracker creates and inits a TxTracker
-func newTxTracker(tx types.Transaction, isClaim bool, counters state.ZKCounters, constraints batchConstraints, weights batchResourceWeights) (*TxTracker, error) {
+// batchResourceWeightMultipliers is a struct that contains the weight multipliers for each resource
+type batchResourceWeightMultipliers struct {
+	cumulativeGasUsed float64
+	arithmetics       float64
+	binaries          float64
+	keccakHashes      float64
+	memAligns         float64
+	poseidonHashes    float64
+	poseidonPaddings  float64
+	steps             float64
+	batchBytesSize    float64
+}
+
+// batchConstraints represents the constraints for a batch in float64
+type batchConstraintsFloat64 struct {
+	maxTxsPerBatch       float64
+	maxBatchBytesSize    float64
+	maxCumulativeGasUsed float64
+	maxKeccakHashes      float64
+	maxPoseidonHashes    float64
+	maxPoseidonPaddings  float64
+	maxMemAligns         float64
+	maxArithmetics       float64
+	maxBinaries          float64
+	maxSteps             float64
+}
+
+// newTxTracker creates and inti a TxTracker
+func newTxTracker(tx types.Transaction, isClaim bool, counters state.ZKCounters, constraints batchConstraints, weights batchResourceWeights, resourceCostMultiplier float64) (*TxTracker, error) {
 	addr, err := state.GetSender(tx)
 	if err != nil {
 		return nil, err
 	}
 
 	txTracker := &TxTracker{
-		Hash:       tx.Hash(),
-		From:       addr,
-		Nonce:      tx.Nonce(),
-		Gas:        tx.Gas(),
-		GasPrice:   tx.GasPrice(),
-		Cost:       tx.Cost(),
-		ReceivedAt: time.Now(),
+		IsClaim:  isClaim,
+		Hash:     tx.Hash(),
+		HashStr:  tx.Hash().String(),
+		From:     addr,
+		FromStr:  addr.String(),
+		Nonce:    tx.Nonce(),
+		Gas:      tx.Gas(),
+		GasPrice: tx.GasPrice(),
+		Cost:     tx.Cost(),
+		BatchResources: batchResources{
+			bytes:      tx.Size(),
+			zKCounters: counters,
+		},
+		resourceCostMultiplier: resourceCostMultiplier,
+		ReceivedAt:             time.Now(),
 	}
 
-	txTracker.IsClaim = isClaim
-	txTracker.BatchResources.zKCounters = counters
-	txTracker.BatchResources.bytes = uint64(tx.Size())
-	txTracker.HashStr = txTracker.Hash.String()
-	txTracker.FromStr = txTracker.From.String()
 	txTracker.Benefit = new(big.Int).Mul(new(big.Int).SetUint64(txTracker.Gas), txTracker.GasPrice)
-	txTracker.calculateEfficiency(constraints, weights)
+	txTracker.constraints = convertBatchConstraintsToFloat64(constraints)
```

Comment:
We are calling convertBatchConstraintsToFloat64, calculating the txTracker.totalWeight and calculateWeightMultipliers each time we are calling newTxTracker, when the result of this functions/calculation is always the same (same values for all the Tx). I think it will more optimal to calculate this in the worker side (and calculate only one time) and then pass the structs/values when calling newTxTracker